action/action-util/src/main/java/com/powsybl/action/util/GeneratorModificationTask.java

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.action.util;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.contingency.tasks.ModificationTask;
+import com.powsybl.iidm.network.Generator;
+import com.powsybl.iidm.network.Network;
+
+import java.util.Objects;
+
+/**
+ * @author Olivier Perrin <olivier.perrin at rte-france.com>
+ */
+public class GeneratorModificationTask implements ModificationTask {
+
+    private final String generatorId;
+    private final Modifs modifs;
+
+    public GeneratorModificationTask(String generatorId, Modifs modifs) {
+        this.generatorId = Objects.requireNonNull(generatorId);
+        this.modifs = modifs;
+    }
+
+    @Override
+    public void modify(Network network, ComputationManager computationManager) {
+        Generator g = network.getGenerator(generatorId);
+        if (g == null) {
+            throw new PowsyblException("Generator '" + generatorId + "' not found");
+        }
+        if (modifs.getMinP() != null) {
+            g.setMinP(modifs.getMinP());
+        }
+        if (modifs.getMaxP() != null) {
+            g.setMaxP(modifs.getMaxP());
+        }
+
+        if (modifs.getVoltageRegulatorOn() != null) {
+            g.setVoltageRegulatorOn(modifs.getVoltageRegulatorOn());
+        }
+
+        boolean skipOtherConnectionChange = false;
+        if (modifs.getConnected() != null) {
+            changeConnectionState(g, modifs.getConnected());
+            skipOtherConnectionChange = true;
+        }
+
+        if (modifs.getTargetP() != null) {
+            setTargetpWithinBoundaries(g, modifs.getTargetP(), skipOtherConnectionChange);
+        } else if (modifs.getpDelta() != null) {
+            setTargetpWithinBoundaries(g, g.getTargetP() + modifs.getpDelta(), skipOtherConnectionChange);
+        }
+
+        if ((modifs.getTargetV() != null) && (g.isVoltageRegulatorOn())) {
+            // The target voltage can be defined only when the generator is in voltage regulation mode.
+            g.setTargetV(modifs.getTargetV());
```

Comment:
Open question : not sure it's useful to be that strict : maybe if a subsequent action connects this generator we'll be suprised the target V is not updated ?

Reply:
Agree, targetV can be set even if only used later (or never...). I would also remove the  g.isVoltageRegulatorOn() check.